demos/http3/ossl-nghttp3-demo-server.c

```diff
@@ -0,0 +1,735 @@
+     * Make sure we won't block as we use select().
+     */
+    if (!SSL_set_blocking_mode(listener, 0))
+        goto err;
+
+    for (;;) {
+        nghttp3_conn *h3conn;
+        nghttp3_settings settings;
+        nghttp3_callbacks callbacks = {0};
+        struct h3ssl h3ssl;
+        const nghttp3_mem *mem = nghttp3_mem_default();
+        int hassomething = 0;
+        nghttp3_nv resp[] = {
+            MAKE_NV(":status", "200"),
+            MAKE_NV("content-length", "20"),
+        };
+        nghttp3_data_reader dr;
+
+        fprintf(stderr, "waiting on socket\n");
+        fflush(stderr);
+        waitsocket(fd, 0);
+        fprintf(stderr, "before SSL_accept_connection\n");
+        fflush(stderr);
+
+        /*
+         * SSL_accept_connection will return NULL if there is nothing to accept
+         */
+        conn = SSL_accept_connection(listener, 0);
+        fprintf(stderr, "after SSL_accept_connection\n");
+        fflush(stderr);
+        if (conn == NULL) {
+            fprintf(stderr, "error while accepting connection\n");
+            continue;
+        }
+
+        /* set the incoming stream policy to accept */
+        if (!SSL_set_incoming_stream_policy(
+                conn, SSL_INCOMING_STREAM_POLICY_ACCEPT, 0)) {
+            fprintf(stderr, "error while setting inccoming stream policy\n");
+            goto err;
+        }
+
+        /*
+         * Service the connection. In a real application this would be done
+         * concurrently. In this demonstration program a single connection is
+         * accepted and serviced at a time.
+         */
+
+        /* try to use nghttp3 to send a response */
+        init_ids(&h3ssl);
+        nghttp3_settings_default(&settings);
+
+        /* Setup callbacks. */
+        callbacks.recv_header = on_recv_header;
+        callbacks.end_headers = on_end_headers;
+        callbacks.recv_data = on_recv_data;
+        callbacks.end_stream = on_end_stream;
+
+        if (nghttp3_conn_server_new(&h3conn, &callbacks, &settings, mem,
+                                    &h3ssl)) {
+            fprintf(stderr, "nghttp3_conn_client_new failed!\n");
+            exit(1);
+        }
+
+        /* add accepted SSL conn to the ids we will poll */
+        add_id(-1, conn, &h3ssl);
+        printf("process_server starting...\n");
+        fflush(stdout);
+
+        /* wait until we have received the headers */
+        while (!h3ssl.end_headers_received) {
+            if (!hassomething)
+                if (waitsocket(fd, 5)) {
+                    printf("read_from_ssl_ids timeout\n");
+                    goto err;
+                }
+            hassomething = read_from_ssl_ids(h3conn, &h3ssl);
+            if (hassomething == -1) {
+                fprintf(stderr, "read_from_ssl_ids hassomething failed\n");
+                goto err;
+            } else if (hassomething == 0) {
+                printf("read_from_ssl_ids hassomething nothing...\n");
+                break;
```

Comment:
If the client does not send any headers, we get an assertion failure from `nghttp3_conn_submit_response()` below. I think this should be a `continue`, not a `break`, with the `while (...)` test deciding when to break.
```suggestion
                continue;
```

Reply:
Oops in fact the hassomething is now just for printf() debugging.

Reply:
Fixed and arranged response sending after testing with chrome and firefox.